js/core/disk/disk-interface.js

```diff
@@ -0,0 +1,72 @@
+// Copyright 2016-present runtime.js project authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+const privates = new WeakMap();
+
+class DiskDriver {
+  constructor(name = '', init = {}) {
+    privates.set(this, {
+      name,
+    });
+    this.onread = null;
+    this.onwrite = null;
+    this.ongetformatinfo = null;
+    this.ongetonline = null;
+    if (typeof init === 'object') {
```

Comment:
Well, `init` could also be a number or a string or anything else truthy if you do it that way. I'll just add `&& init !== null`.